libs/core/src/service/file_service.rs

```diff
@@ -23,7 +23,10 @@ impl<Client: Requester> CoreState<Client> {
         let id =
             tree.create(Uuid::new_v4(), symkey::generate_key(), parent, name, file_type, account)?;
 
-        let ui_file = tree.finalize(&id, account, &mut self.db.pub_key_lookup)?;
+        let mut ui_file = tree.finalize(account, id, &mut self.db.pub_key_lookup)?;
+        if matches!(file_type, FileType::Link { .. }) {
+            ui_file.id = id;
+        }
```

Comment:
I think after responding to other feedback, this (and other instances below) won't be necessary

Reply:
i think that's still needed because when creating a link, the link's id, instead of the target id should be returned. This mirrors the behavior that happens at [create_link_at_path](https://github.com/lockbook/lockbook/pull/1583/files#diff-2b163d75063f996c29e322b77ef8ccea13fc40d82e5220bc27c5fe8ad470a014R24)

Reply:
This doesn't do anything tho? You decrypt a particular id, and assign it's id to the same id.